util/crs-rules-check/rules-check.py

```diff
@@ -850,7 +850,7 @@ def generate_version_string():
         # if no --version/-v was given, get version from git describe --tags output
         crsversion = generate_version_string()
```

Comment:
The script has a `generate_version_string()` method which produces a similar version string that you make outside of the script, [here](https://github.com/coreruleset/coreruleset/blob/c57c458da6158cd747c0e4c193de9729748de67f/util/crs-rules-check/rules-check.py#L811-L827). May be we should replace the logic there by yours. Or could we leave it as is? What do you think?

Reply:
Yes, we could do that, but then we would need to pass in the base ref (the "target branch"). And even then, part of the logic will continue to live in the pipeline (fetching of the base ref, tags, etc.). The only logic that makes sense to have in the script, IMO, is the creation of the expected version string from the tag.

Something like this:
1. if `-v` is empty, don't perform the version check
2. if `-v` is not empty, `args.version` contains the Git tag
3. if `-v` is not empty, convert Git tag to expected version and run check

I'd then also rename `-v` to `check-version`, for example.